code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -0,0 +1,526 @@
+	response_disarm = "flails at"
+	response_harm   = "punches"
+	icon = 'icons/mob/mob.dmi'
+	icon_state = "stand"
+	icon_living = "stand"
+	speed = 0
+	a_intent = "harm"
+	stop_automated_movement = 1
+	floating = 1
+	attack_sound = 'sound/weapons/punch1.ogg'
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	attacktext = "punches"
+	maxHealth = 100000 //The spirit itself is invincible
+	health = 100000
+	environment_smash = 0
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	butcher_results = list(/obj/item/weapon/ectoplasm = 1)
+	var/cooldown = 0
+	var/damage_transfer = 1 //how much damage from each attack we transfer to the owner
+	var/mob/living/summoner
+	var/range = 10 //how far from the user the spirit can be
+	var/playstyle_string = "You are a standard Guardian. You shouldn't exist!"
+	var/magic_fluff_string = " You draw the Coder, symbolizing bugs and errors. This shouldn't happen! Submit a bug report!"
+	var/tech_fluff_string = "BOOT SEQUENCE COMPLETE. ERROR MODULE LOADED. THIS SHOULDN'T HAPPEN. Submit a bug report!"
+	var/bio_fluff_string = "Your scarabs fail to mutate. This shouldn't happen! Submit a bug report!"
+
+
+/mob/living/simple_animal/hostile/guardian/Life() //Dies if the summoner dies
+	..()
+	if(summoner)
+		if(summoner.stat == DEAD)
+			src << "<span class='danger'>Your summoner has died!</span>"
+			visible_message("<span class='danger'>The [src] dies along with its user!</span>")
+			ghostize()
+			qdel(src)
+	else
+		src << "<span class='danger'>Your summoner has died!</span>"
+		visible_message("<span class='danger'>The [src] dies along with its user!</span>")
+		ghostize()
+		qdel(src)
+	if(summoner)
+		if (get_dist(get_turf(summoner),get_turf(src)) <= range)
+			return
+		else
+			src << "You moved out of range, and were pulled back! You can only move [range] meters from [summoner.real_name]"
+			visible_message("<span class='danger'>The [src] jumps back to its user.</span>")
+			loc = get_turf(summoner)
+
+/mob/living/simple_animal/hostile/guardian/Move() //Returns to summoner if they move out of range
+	..()
+	if(summoner)
+		if (get_dist(get_turf(summoner),get_turf(src)) <= range)
+			return
+		else
+			src << "You moved out of range, and were pulled back! You can only move [range] meters from [summoner.real_name]"
+			visible_message("<span class='danger'>The [src] jumps back to its user.</span>")
+			loc = get_turf(summoner)
+
+
+/mob/living/simple_animal/hostile/guardian/adjustBruteLoss(amount) //The spirit is invincible, but passes on damage to the summoner
+	var/damage = amount * src.damage_transfer
+	if (src.summoner)
+		src.summoner.adjustBruteLoss(damage)
+		if(damage)
+			src.summoner << "<span class='danger'><B>Your [src.name] is under attack! You take damage!</span></B>"
+
+
+/mob/living/simple_animal/hostile/guardian/ex_act(severity, target)
+	switch (severity)
+		if (1.0)
```

Comment:
remove the ```.0``` it's found in other procs because it's a decompilation artefact of ```1``` --> ```1.0```